.github/workflows/testingWorkflow.yml

```diff
@@ -14,25 +22,220 @@ on:
+          context: ./
+          file: ./frontend/Dockerfile
+          acr_password: ${{ secrets.ACR_ADMIN_PASWORD }}
+          acr_registry: ${{ secrets.ACR_REPO_URL }}
+          acr_username: ${{ secrets.ACR_ADMIN_USERNAME }}
+          gh_registry: ghcr.io
+          gh_username: ${{ github.actor }}
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+          image_name: frontend
+          platform: ${{ matrix.platform }}
 
   build_frontend_lighthouse_image:
-    uses: ./.github/workflows/buildFrontendLighthouseImage.yml
+    if: needs.frontend_changes.outputs.has_changes == 'true' || inputs.force_build == 'true' || github.ref == 'refs/heads/main'
+    needs:
+      - frontend_changes
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - platform: ${{ inputs.platform }}
+    outputs:
+      version: ${{ steps.set_frontend_lighthouse_version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build Frontend Lighthouse Image
+        id: set_frontend_lighthouse_version
+        uses: ./.github/actions/build-docker-image
+        with:
+          context: ./
+          file: ./frontend/Dockerfile
+          acr_password: ${{ secrets.ACR_ADMIN_PASWORD }}
+          acr_registry: ${{ secrets.ACR_REPO_URL }}
+          acr_username: ${{ secrets.ACR_ADMIN_USERNAME }}
+          gh_registry: ghcr.io
+          gh_username: ${{ github.actor }}
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+          image_name: frontend-lighthouse
+          platform: ${{ matrix.platform }}
+          build_args: |
+            "REACT_APP_BASE_URL=https://localhost.simplereport.gov"
+            "REACT_APP_BACKEND_URL=https://localhost.simplereport.gov/api"
+            "PUBLIC_URL=/app/"
+            "REACT_APP_OKTA_ENABLED=false"
+            "REACT_APP_DISABLE_MAINTENANCE_BANNER=true"
 
   build_nginx_image:
-    uses: ./.github/workflows/buildNginxImage.yml
+    if: needs.nginx_changes.outputs.has_changes == 'true' || inputs.force_build == 'true' || github.ref == 'refs/heads/main'
+    needs:
+      - nginx_changes
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - platform: ${{ inputs.platform }}
+    outputs:
+      version: ${{ steps.set_nginx_version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build Nginx Image
+        id: set_nginx_version
+        uses: ./.github/actions/build-docker-image
+        with:
+          context: ./nginx
+          acr_password: ${{ secrets.ACR_ADMIN_PASWORD }}
+          acr_registry: ${{ secrets.ACR_REPO_URL }}
+          acr_username: ${{ secrets.ACR_ADMIN_USERNAME }}
+          gh_registry: ghcr.io
+          gh_username: ${{ github.actor }}
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+          image_name: nginx
+          platform: ${{ matrix.platform }}
 
+# Automated tests
   e2e_local:
+    if: |
+      always() &&
```

Comment:
I tried to cancel the e2e_local job but it doesn't work -- I think because of this? https://github.com/actions/runner/issues/1846#issuecomment-1246102753

Do we want these jobs to be not cancelable? 🤔

Reply:
:thinking:

Reply:
This is a great callout. I think removing `always()` would be completely fine and doesn't add anything to this PR. I'll remove it, give it a test, and update here with the results.

Reply:
Canceling works again, and I'm not seeing any other issues with removing `always()`. Thanks again!

Reply:
@emyl3 I overlooked something; because of the way we skip some jobs, I'll have to adjust to this. A bit of churn here, but I appreciate the callout.